server/src/main/java/org/elasticsearch/index/analysis/AnalyzerProvider.java

```diff
@@ -12,12 +12,12 @@
 import org.apache.lucene.analysis.Analyzer;
 import org.elasticsearch.injection.guice.Provider;
 
-public interface AnalyzerProvider<T extends Analyzer> extends Provider<T> {
+public interface AnalyzerProvider<T extends Analyzer> extends Provider<Analyzer> {
```

Comment:
Is this change required?

Reply:
Yes, because PersianAnalyzerProvider now either creates a PersianAnalyzer or an alternative StopwordAnalyzerBase implementation based on the index version. Since PersianAnalyzer (like almost all other analyzers in Lucene) is final, we cannot change any of its behavior. We don't need the interface type anywhere anyway, so we can simply user Analyzer here.

Reply:
I still don't get it, sorry. It looks odd to me that we need to keep the generic type yet specify `Provider<Analyzer>`, perhaps there's something I am missing.

Reply:
That option didn't occur to me, probably because `StopwordAnalyzerBase` is abstract but you're right, that's an option I will try. 
And yes, I would have liked to remove the generic Type from AnalyzerProvider entirely because it seems unused but that would indeed be a larger change (though possibly mostly mechanical), so I avoided that.

Reply:
See [57b0686](https://github.com/elastic/elasticsearch/pull/113482/commits/57b068610229c445f729489534d8cdf23d49561b) for an update